packages/salesforcedx-vscode-core/src/commands/templates/internalCommandUtils.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer,
+  PostconditionChecker,
+  PreconditionChecker
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { Uri } from 'vscode';
+import { channelService } from '../../channels';
+import { notificationService } from '../../notifications';
+import { sfdxCoreSettings } from '../../settings';
+import { telemetryService } from '../../telemetry';
+import { AURA_METADATA_TYPE, LWC_METADATA_TYPE } from './metadataTypeConstants';
+
+export class InternalDevWorkspaceChecker implements PreconditionChecker {
+  public check(): boolean {
+    return sfdxCoreSettings.getInternalDev();
+  }
+}
+
+export class FileInternalPathGatherer
```

Comment:
The old approach of presenting a list of available folders, even when filtering to exclusively the valid ones, was not performant. Got rid of that code and replace it with this one which takes the file path from the context of the user right-clicking a folder.

FYI: I might add one more validation here, depending on some tests I'm running.

Reply:
Will you still be able to run these form the command palette or is it required now that you right click a folder?

Reply:
No, we won't do command palette in this release. I didn't find a performant way for the commands to work using the command palette in a big project even after ignoring invalid folder paths.